addons/website_slides/static/src/js/slides_course_join.js

```diff
@@ -1,28 +1,65 @@
 odoo.define('website_slides.course.join.widget', function (require) {
 'use strict';
-
-var core = require('web.core');
+    
 var publicWidget = require('web.public.widget');
-require('website_slides.slides');
+var core = require('web.core');
 
 var _t = core._t;
 
 var CourseJoinWidget = publicWidget.Widget.extend({
-    template: 'slide.course.join',
+    template: 'slide.slide.quiz.join',
     xmlDependencies: ['/website_slides/static/src/xml/channel_management.xml'],
     events: {
-        'click .o_wslides_js_course_join_link': '_onClickJoin',
-    },
-
-    init: function (parent, channelId){
-        this.channelId = channelId;
-        return this._super.apply(this, arguments);
+        'click .o_wslides_js_course_join_link': '_onClickJoin'
+    },  
+    init: function (parent, options) {
+        this._super.apply(this, arguments);
+        this.quizWidget = parent || false;
+        this.quiz = options.quiz || false;
+        this.signupAllowed = options.channel.signupAllowed || false;
+        this.channelId = options.channel.channelId || false;
+        this.channelEnroll = options.channel.channelEnroll || false;
+        this.isMember = options.isMember || false;
+        this.publicUser = options.publicUser || false;
+        this.slide = options.slide || false;
     },
 
     //--------------------------------------------------------------------------
     // Private
     //--------------------------------------------------------------------------
-
+    /**
+     * @private
+     */
+    _createLoginRedirectUrl: function () {
+        var urlParts = window.location.pathname.split('/');
+        var url = urlParts.join('/');
+        var params = {};
+        if (window.location.href.indexOf("fullscreen") > -1) {
+            params.fullscreen = 1;
+        }
+        this.redirectURL = _.str.sprintf('%s?%s', url, $.param(params));
+        return _.str.sprintf('/web/login?redirect=%s', encodeURIComponent(this.redirectURL));
+    },
+    /**
+     * @private
+     * @param {MouseEvent} event
+     */
+    _joinCourse: function () {
```

Comment:
this method is useless.
Do directly `this.joinChannel(this.channelId);` inside `_onClickJoin`